app/tests/spec/views/elements/token-code-input.js

```diff
@@ -0,0 +1,79 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const $ = require('jquery');
+const {assert} = require('chai');
+const AuthErrors = require('lib/auth-errors');
+const TotpEl = require('views/elements/token-code-input');
+
+const TEMPLATE =
+  '<input type="number" pattern="\d*" class="token-code"></input>' +
+  '<input type="number" pattern="\d*" class="not-code"></input>';
+
+describe('views/elements/token-code-input', function () {
+  let $element;
+  let $otherElement;
+
+  before(() => {
+    $('#container').html(TEMPLATE);
+    $element = $('.token-code');
+    $otherElement = $('.not-code');
```

Comment:
Fwiw, I'm not a fan of these unit tests that rely on DOM manipulation because unit tests are supposed to be fast and modifying the DOM is slow. But I know we do it in lots of places so I'm just raising the objection out of principle, not expecting anything to actually come from it.